### PR TITLE
Enhancement/5229 Add Sharing Settings Permissions Tests

### PR DESCRIPTION
**Pull request by jimmymadon**
## Summary

Addresses issue:

- #5229 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: tests/phpunit/integration/Core/Modules/REST_Dashboard_Sharing_ControllerTest.php [hunk 1]
```diff
@@ -80,81 +80,219 @@ public function test_register() {
...
+				// Default settings always saved for shared ownership modules.
+				'pagespeed-insights' => array(
+					'sharedRoles' => array(),
+					'management'  => 'all_admins',
+				),
 			),
-			'pagespeed-insights' => array(
-				'sharedRoles' => array( 'editor', 'subscriber' ),
-				'management'  => 'all_admins',
+			'newOwnerIDs' => (object) array(
```
Comment (tofumatt) [inline tests/phpunit/integration/Core/Modules/REST_Dashboard_Sharing_ControllerTest.php:149]: Why is the `object` cast needed here? Just seems a bit odd to me.

```suggestion
			'newOwnerIDs' => array(
```

File: tests/phpunit/integration/Core/Modules/REST_Dashboard_Sharing_ControllerTest.php [hunk 2]
```diff
@@ -80,81 +80,219 @@ public function test_register() {
...
+						'management'  => 'all_admins',
+					),
+					'analytics'      => array(
+						'sharedRoles' => array( 'editor' ),
+					),
+				),
+			)
+		);
+		$response = rest_get_server()->dispatch( $request );
+		$this->assertEquals( 'rest_forbidden', $response->get_data()['code'] );
```
Comment (tofumatt) [inline tests/phpunit/integration/Core/Modules/REST_Dashboard_Sharing_ControllerTest.php:105]:
```suggestion
		// This admin hasn't authenticated with the Site Kit proxy service yet,
		// so they aren't allowed to modify Dashboard Sharing settings.
		$this->assertEquals( 'rest_forbidden', $response->get_data()['code'] );
```

File: tests/phpunit/integration/Core/Modules/REST_Dashboard_Sharing_ControllerTest.php [hunk 3]
```diff
@@ -174,6 +312,7 @@ private function grant_manage_options_permission() {
 		);
 
 		// Re-register Permissions after enabling the dashboardSharing feature to include dashboard sharing capabilities.
+		// TODO Remove this when dashboardSharing feature flag is removed.
```
Comment (tofumatt) [inline tests/phpunit/integration/Core/Modules/REST_Dashboard_Sharing_ControllerTest.php:315]:
```suggestion
		// TODO: Remove this when `dashboardSharing` feature flag is removed.
```

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks good to me, thanks for all the tests!

Just a few small tweaks/questions then I think this is good-to-go 👍🏻

File: tests/phpunit/integration/Core/Modules/REST_Dashboard_Sharing_ControllerTest.php (hunk 1 above)
Comment (jimmymadon) [reply tests/phpunit/integration/Core/Modules/REST_Dashboard_Sharing_ControllerTest.php:149]: Odd indeed! But it is to [maintain the shape of the response](https://github.com/google/site-kit-wp/issues/4481#issuecomment-1130393863) when translated to JSON.

**Review by tofumatt — APPROVED** [review]
Looks good 👍🏻

**Pull request merged**